components/kvengine/src/table/sstable/iterator.rs

```diff
@@ -64,12 +65,13 @@ impl BlockIterator {
 
     fn load_entries(&mut self) {
         let data = self.b.chunk();
-        let num_entries = LittleEndian::read_u32(data) as usize;
+        assert_eq!(data[0], BLOCK_FORMAT_V1);
+        let num_entries = LittleEndian::read_u32(&data[1..]) as usize;
         self.entry_offs = unsafe {
-            let ptr = data[4..].as_ptr() as *mut u32;
+            let ptr = data[5..].as_ptr() as *mut u32;
```

Comment:
It's not memory-aligned. I'm afriad it has poor performance. I prefer to use `u32` as version.

BTW, `finish_block()` uses `put_u32_le()` to write offsets, why here casts it to u32 directly?